var/spack/repos/builtin/packages/py-jupyterlab/package.py

```diff
@@ -34,23 +35,29 @@ class PyJupyterlab(PythonPackage):
     depends_on("npm", type="run")
 
     depends_on("python@3.8:", when="@4:", type=("build", "run"))
-    depends_on("py-hatchling@1.5:", when="@4:", type=("build", "run"))
+    depends_on("py-hatchling@1.21.1:", when="@4.3.5:", type="build")
+    depends_on("py-hatchling@1.5:", when="@4:", type="build")
     # under [tool.hatch.build.hooks.jupyter-builder] in pyproject.toml
     depends_on("py-hatch-jupyter-builder@0.3.2:", when="@4:", type=("build", "run"))
 
     depends_on("py-async-lru@1:", when="@4:", type=("build", "run"))
+    depends_on("py-httpx@0.25:", when="@4.3.5:", type=("build", "run"))
     depends_on("py-importlib-metadata@4.8.3:", when="@4: ^python@:3.9", type=("build", "run"))
     depends_on("py-importlib-resources@1.4:", when="@4: ^python@:3.8", type=("build", "run"))
+    depends_on("py-ipykernel@6.5.0:", when="@4.3.5:", type=("build", "run"))
     depends_on("py-ipykernel", when="@4:", type=("build", "run"))
     depends_on("py-jinja2@3.0.3:", when="@4:", type=("build", "run"))
     depends_on("py-jupyter-core", when="@3:", type=("build", "run"))
     depends_on("py-jupyter-lsp@2:", when="@4:", type=("build", "run"))
     depends_on("py-jupyter-server@2.4:2", when="@4:", type=("build", "run"))
+    depends_on("py-jupyterlab-server@2.27.1:2", when="@4.3.5:", type=("build", "run"))
     depends_on("py-jupyterlab-server@2.19:2", when="@4:", type=("build", "run"))
```

Comment:
Fix the ordering here (newer first). Do you know why there are two different ranges for `@4:` at lines 52 and 54?

Reply:
Will fix the ordering! I think lines 52 and 54 are different packages with similar names, the difference being `jupyter` vs `jupyterlab`.

Reply:
🤦 You're right about the different packages. Not sure how I missed that.

Reply:
The ordering here doesn't consistently match that in the `pyproject.toml` file.

Reply:
Actually maybe the ordering is okay? Line 53 is newer than 54, and 52 is for a different dep. Is that what you meant, or should I lift all the 4.3.5 versions higher in the package?

Reply:
Yes, it is okay.